src/Donation/PayboxPaymentUnsubscription.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace AppBundle\Donation;
+
+use AppBundle\Entity\Adherent;
+use AppBundle\Entity\Donation;
+use AppBundle\Exception\PayboxPaymentUnsubscriptionException;
+use AppBundle\Mailer\MailerService;
+use AppBundle\Mailer\Message\PayboxPaymentUnsubscriptionConfirmationMessage;
+use Lexik\Bundle\PayboxBundle\Paybox\System\Cancellation\Request;
+
+class PayboxPaymentUnsubscription
+{
+    private $request;
+    private $donationRequestUtils;
+    private $mailer;
+
+    public function __construct(MailerService $mailer, Request $request, DonationRequestUtils $donationRequestUtils)
+    {
+        $this->mailer = $mailer;
+        $this->request = $request;
+        $this->donationRequestUtils = $donationRequestUtils;
+    }
+
+    /**
+     * @throws PayboxPaymentUnsubscriptionException
+     */
+    public function unsubscribe(Donation $donation): void
+    {
+        $result = $this->request->cancel($this->donationRequestUtils->buildDonationReference($donation));
```

Comment:
why do you not log when one error occurred?

Reply:
Witch error do you think ?